### ISSUE TITLE
[package] qt/6.x: is cross-building possible?

### ISSUE THREAD

**Issue by Bueddl** [conan-io/conan-center-index#5018]
### Package and Environment Details
  * Package Name/Version: **qt/6.x**
#### Builder
  * Operating System+version: **Linux Debian stable**
  * Compiler+version: **Clang 11.0.1**
  * Docker image: **debian/stable**
  * Conan version: **conan 1.34.0**
#### Host
  * Operating System+version: **Windows x86_64**
  * Cross-Compiler+version: **Clang 11.0.1** (wrapped using wclang, https://github.com/tpoechtrager/wclang, to conform to the mingw interface)

#### Build
```
Configuration for profile /home/user/tmp/conan/host-linux-x86_64.conf:

[settings]
cppstd=17
os=Linux
arch=x86_64
os_build=Linux
arch_build=x86_64
compiler=clang
compiler.version=11
compiler.libcxx=libstdc++11
[options]
[build_requires]
[env]
CONAN_CMAKE_TOOLCHAIN_FILE=/home/user/tmp/cmake/Toolchain-host-linux.cmake
CC=clang
CXX=clang++
LD=clang++
LDFLAGS=-fuse-ld=lld
libtool:CC=gcc
libtool:CXX=g++
libtool:LD=
libtool:LDFLAGS=
```

(I left the libtool specific stuff in here since it was build as a dependency for the builder qt and failed if I didn't link using the binutils linker which is picked by default when using gcc).

#### Host
```
Configuration for profile /home/user/tmp/conan/cross-windows-x86_64.conf:

[settings]
cppstd=17
os=Windows
arch=x86_64
os_build=Linux
arch_build=x86_64
compiler=clang
compiler.version=11
compiler.libcxx=libstdc++11
[options]
[build_requires]
[env]
CONAN_CMAKE_FIND_ROOT_PATH=/usr/x86_64-w64-mingw32/
CONAN_CMAKE_SYSROOT=/usr/x86_64-w64-mingw32/
CONAN_CMAKE_TOOLCHAIN_FILE=/home/user/tmp/cmake/Toolchain-cross-windows.cmake
CHOST=x86_64-w64-mingw32
AR=x86_64-w64-mingw32-ar
AS=x86_64-w64-mingw32-as
RANLIB=x86_64-w64-mingw32-ranlib
CC=x86_64-w64-mingw32-clang
CXX=x86_64-w64-mingw32-clang++
STRIP=x86_64-w64-mingw32-strip
RC=x86_64-w64-mingw32-windres
LD=x86_64-w64-mingw32-clang++
LDFLAGS=-fuse-ld=lld -Wl,/subsystem:windows -Wl,/dynamicbase -Wl,/fixed:no -Wl,/nxcompat -Wl,/highentropyva
```

### Description / Steps to reproduce

I am trying to get a cross build for qt/6.x working from Linux:x86_64 to Windows:x86_64. By default that doesn't seem to be possible for qt/6.x yet since with the transition from 5.x to 6.x qt requires the user to provide the build tools via env `QT_HOST_PATH` instead of building the required tools for the build-machine itself (unsure about this but this is how it looks to me at the moment).

I tried to get around this by patching the `conanfile.py` myself and add qt as a build-requirement when cross-building. However this doesn't work since, if I understand correctly, conan will wipe out the build directory of the build for the builder-machine when it gets to build qt for the host machine since it will pick the same directory name for those two builds. 
Additionally things get worse since both qt builds will share the same options and thus lots of not required targets will be compiled for the builder-machine.

I could, in theory, compile qt locally without conan for the builder (or install a prebuilt version of it) and specify `QT_HOST_PATH` but that comes with the drawback that I need to keep the conan version and the local version synced and I somehow loose the benefit from a package manager.

Is there any way to cross-build qt/6.x using conan?
</details>

Thank you!

**Comment by anton-danielsson** [comment]
Yes, as you have seen Conan seems to have some issues when the same recipe is in both host and build context.
I have seen similar problems in https://github.com/conan-io/conan/issues/8443#issuecomment-774192351
Otherwise I think it's a good approach to depend on yourself to provide build tools but until the problems
are fixed in Conan it will be difficult.
Adding a separate "qt-build-tools" recipe should work but I'm afraid it might lead to a lot of code duplication...

**Comment by perseoGI** [comment]
Hi there!
`qt` recipe has suffered plenty of modifications since this issue was open.
As far as I know, cross compilation has been successfully addressed.

I'm closing this issue!
If the problem remains, do not hesitate to open a new ticket with updated information, preferably using conan v2.
Happy coding 🐸
